lib/postgrex/notifications.ex

```diff
@@ -102,19 +102,19 @@ defmodule Postgrex.Notifications do
     ref = Process.monitor(pid)
 
     s = put_in(s.listeners[ref], {channel, pid})
-    s = update_in(s.listener_channels[channel], &((&1 || HashSet.new()) |> HashSet.put(ref)))
+    s = update_in(s.listener_channels[channel], &((&1 || MapSet.new()) |> MapSet.put(ref)))
```

Comment:
Maybe we could have `Map.put(ref, pid)`? The `ref` would need to be the key I think. A `MapSet` basically just a map where value is ignore.